src/Nova-GPU-Driver.md

```diff
@@ -1,62 +1,92 @@
-infrastructure in other subsystems (notably PCI and DRM).
+Nova is a driver for GSP (global system processor) based Nvidia GPUs. It is
+intended to become the successor of Nouveau as the mainline driver for Nvidia
+(GSP) GPUs on Linux.
 
 ## Contact
 
-To contact the team and / or participate in development, please use the mailing
-list: nouveau@lists.freedesktop.org
+Available communication channels are:
+
+- The mailing list: nouveau@lists.freedesktop.org
+- IRC: #nouveau on OFTC
+- [Zulip Chat](https://rust-for-linux.zulipchat.com/#narrow/channel/509436-Nova)
 
 
 ## Resources
 
-- [Official Source Tree](https://gitlab.freedesktop.org/drm/nova)
-- [Announcement E-Mail](https://lore.kernel.org/dri-devel/Zfsj0_tb-0-tNrJy@cassiopeiae/)
+The parts that are already in mainline Linux can be found in
+`drivers/gpu/nova-core/` and `drivers/gpu/drm/nova/`
+
+The out-of-tree repository where active development takes place is located on
+[Freedesktop](https://gitlab.freedesktop.org/drm/nova).
+
+
+## Background
+
+### Why a new driver?
+
+Nouveau was, for the most part, designed for pre-GSP hardware. The driver exists
+since ~2009 and its authors back in the day had to reverse engineer a lot about
+the hardware's internals, resulting in a relatively difficult to maintain
+codebase.
+
+Moreover, Nouveau's maintainers concluded that a new driver, exclusively for
+(new) GSP hardware, would allow for significantly simplifying the driver
+design.
 
-In the source tree, the driver lives in `drivers/gpu/drm/nova`.
 
+### Why write it in Rust?
 
-## Status
+Rust's most desired feature are its guarantees for memory safety, notably the
+elimination of Use-after-Free errors. Those are errors GPU drivers suffer from
+significantly, because GPUs are, by definition, asynchronous in regards to the
+CPU and can handle millions of jobs (i.e., memory objects) simultaneously. Jobs'
+status can be changed at different places in the code base at different
+points in time (through work items, interrupt handlers, userspace calls, ...).
 
-Currently, Nova is just a stub driver intended to lift the bindings necessary
-for a real GPU driver into the (mainline) kernel.
+In short, GPU drivers were expected to profit the most from the promised memory
+safety.
 
-Currently, those efforts are mostly focused on getting bindings for PCI, DRM
-and the Device (driver) model upstream.
+Since Nova was a freshly written new driver, it was an opportunity to try to
+leverage the advantages of Rust and obtain a more reliable, maintainable driver.
 
-It can be expected that, as the driver continues to grow, various other abstractions
-will be needed.
+Another reason is that Rust's macros are helpful when creating firmware bindings
+for continuously changing firmware versions and their ABIs.
 
 
-## Utilized Common Rust Infrastructure
+## Architecture
 
-Nova depends on the Rust for Linux `staging/*` [branches](Branches.md).
+![Nova Architecture with vGPUs](./nova-core-vm.png)
```

Comment:
Thanks for the diagram!

I think you're missing the vGPU driver that sits on top of nova-core on the host side. There's also a good diagram that Zhi attached to his initial series for vGPU [1].

[1] https://lore.kernel.org/nouveau/20240922124951.1946072-1-zhiw@nvidia.com/